pkg/asset/installconfig/ovirt/credentials.go

```diff
@@ -1,92 +1,272 @@
+		}
+		logrus.Debugf("Loaded %s into the system pool!", pemFilePath)
+	}
+	return false
+}
+
+// downloadFile from specificed URL and store via filepath
+// Return error in case of failure
+func (c clientHTTP) downloadFile() error {
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: c.skipVerify,
+			RootCAs:            c.certPool,
+		},
+	}
+
+	client := &http.Client{Transport: tr}
+	resp, err := client.Get(c.urlAddr)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if c.saveFilePath == "" {
+		return errors.Wrapf(err, "saveFilePath must be specificed")
+	}
+
+	out, err := os.Create(c.saveFilePath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	return err
+}
+
+// checkURLResponse performs a GET on the provided urlAddr to ensure that
+// the url actually exists. Users can set skipVerify as true or false to
+// avoid cert validation. In case of failure, returns error.
+func (c clientHTTP) checkURLResponse() error {
+
+	logrus.Debugf("Checking URL Response... urlAddr: %s skipVerify: %s", c.urlAddr, strconv.FormatBool(c.skipVerify))
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: c.skipVerify,
+			RootCAs:            c.certPool,
+		},
+	}
+
+	client := &http.Client{Transport: tr}
+	resp, err := client.Get(c.urlAddr)
+	if err != nil {
+		return errors.Wrapf(err, "Error checking URL response")
+	}
+	defer resp.Body.Close()
+
+	return nil
+}
+
+// askPassword will ask the password to connect to Engine API.
+// The password provided will be added in the Config struct.
+// If an error happens, it will ask again username for users.
+func askPassword() error {
 	err := survey.Ask([]*survey.Question{
 		{
-			Prompt: &survey.Input{
-				Message: "oVirt API endpoint URL",
-				Help:    "The URL of the oVirt engine API. For example, https://ovirt-engine-fqdn/ovirt-engine/api.",
+			Prompt: &survey.Password{
+				Message: "Engine password",
+				Help:    "",
 			},
-			Validate: survey.ComposeValidators(survey.Required),
+			Validate: survey.ComposeValidators(survey.Required, authenticated(&EngineConfig)),
 		},
-	}, &c.URL)
+	}, &EngineConfig.Password)
 	if err != nil {
-		return c, err
+		return err
 	}
 
-	var ovirtCertTrusted bool
-	err = survey.AskOne(
-		&survey.Confirm{
-			Message: "Is the oVirt CA trusted locally?",
-			Default: true,
-			Help:    "In order to securly communicate with the oVirt engine, the certificate authority must be trusted by the local system.",
+	return nil
+}
+
+// askUsername will ask username to connect to Engine API.
+// The username provided will be added in the Config struct.
+// Returns Config and error if failure.
+func askUsername() error {
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Engine username",
+				Help:    "The user must have permissions to create VMs and disks on the Storage Domain with the same name as the OpenShift cluster.",
```

Comment:
this help is orthogonal to what this field really is.

Reply:
It's just a copy/past what's already there but I am going to provide a new message.